linkis-commons/linkis-storage/src/main/java/org/apache/linkis/storage/source/FileSplit.java

```diff
@@ -0,0 +1,248 @@
+  }
+
+  public void page(int page, int pageSize) {
+    if (!pageTrigger) {
+      start = (page - 1) * pageSize;
+      end = pageSize * page - 1;
+      pageTrigger = true;
+    }
+  }
+
+  public String getType() {
+    return type;
+  }
+
+  public void addParams(Map<String, String> params) {
+    this.params.putAll(params);
+  }
+
+  public void addParams(String key, String value) {
+    this.params.put(key, value);
+  }
+
+  public Map<String, String> getParams() {
+    return params;
+  }
+
+  public int getTotalLine() {
+    return totalLine;
+  }
+
+  public <M> M whileLoop(Function<MetaData, M> metaDataFunction, Consumer<Record> recordConsumer) {
+    M m = null;
+    try {
+      MetaData metaData = fsReader.getMetaData();
+      m = metaDataFunction.apply(metaData);
+      if (pageTrigger) {
+        fsReader.skip(start);
+      }
+      count = start;
+      boolean hasRemovedFlag = false;
+      while (fsReader.hasNext() && ifContinueRead()) {
+        Record record = fsReader.getRecord();
+        boolean needRemoveFlag = false;
+        if (!hasRemovedFlag && fsReader instanceof StorageScriptFsReader) {
+          Parser parser = ((StorageScriptFsReader) fsReader).getScriptParser();
+          Variable[] meta = ((ScriptMetaData) metaData).getMetaData();
+          if (meta != null
+              && meta.length > 0
+              && parser != null
+              && parser.getAnnotationSymbol().equals(record.toString())) {
+            needRemoveFlag = true;
+            hasRemovedFlag = true;
+          }
+        }
+        if (!needRemoveFlag) {
+          recordConsumer.accept(shuffler.apply(record));
+          totalLine++;
+          count++;
+        }
+      }
+    } catch (IOException e) {
+      logger.warn("FileSplit forEach failed", e);
+    }
+    return m;
+  }
+
+  public void biConsumer(Consumer<MetaData> metaDataFunction, Consumer<Record> recordConsumer) {
+    try {
+      MetaData metaData = fsReader.getMetaData();
+      metaDataFunction.accept(metaData);
+      if (pageTrigger) {
+        fsReader.skip(start);
+      }
+      count = start;
+      while (fsReader.hasNext() && ifContinueRead()) {
+        recordConsumer.accept(shuffler.apply(fsReader.getRecord()));
+        totalLine++;
+        count++;
+      }
+    } catch (IOException e) {
+      logger.warn("FileSplit biConsumer failed", e);
+    }
+  }
+
+  public Pair<Integer, Integer> getFileInfo(int needToCountRowNumber) {
+    int colNumber = 0;
+    int rowNumber = 0;
+    MetaData metaData = null;
+    try {
+      metaData = fsReader.getMetaData();
+      colNumber =
+          metaData instanceof TableMetaData ? ((TableMetaData) metaData).getColumns().length : 1;
+      rowNumber = 0;
+      rowNumber =
+          needToCountRowNumber == -1
+              ? fsReader.skip(Integer.MAX_VALUE)
+              : fsReader.skip(needToCountRowNumber);
+    } catch (IOException e) {
+      logger.warn("FileSplit getFileInfo failed", e);
```

Comment:
An exception should be thrown, optionally LinkisRuntimeException

Reply:
Already throw LinkisRuntimeException here,Thanks.